### DISCUSSION TITLE
r inline code not executed?

### DISCUSSION THREAD

**Discussion by giabaio** [quarto-dev/quarto-cli#4855] [Q&A] [answered]
I'm experiencing a weird behaviour. I have recently updated my computer (but I don't think this should be the issue --- I've scripted my way into re-installing all the relevant packages etc). I have now a `qmd` document to output in `html` as part of a website. Something like
```
---
title: "Second Continuous Module Dialogue"
author: "Gianluca Baio"
date: "2023-03-16 13:14:38"
categories: 
   - "Assessment"
authors: ["Written by Gianluca"]
page-layout: full
toc: false
---

The voting code is **2635** and you have 2 days (from `r Sys.time() |> format("%I:%M%p, %A, %d %B %Y")`) to participate. You can also use this QR.
```

If I compile this, the `R` is simply not executed --- the output is typeset as code but taken "verbatim" so it appears like:
```
r Sys.time() |> format("%I:%M%p, %A, %d %B %Y")
```
instead of being executed to render the actual time/date.

If I place the same command into a full `R` chunk, then things are OK.
Any idea of why I'm seeing this?

**Comment by rgaiacs** [comment]
I remember some conversation of quarto trying to be smart to decide how it should use to convert your `.qmd` to `.md`. One of the things that quarto do is to look for a executable code block like

``````
```{r}
1 + 1
```
``````

If your document doesn't have a executable code block, quarto treats your document as markdown and send it to the next step.

If your document has a executable R code chunk, it will process your document with R Markdown/knitr.

If your document has a executable code chunck in a language other than R, it will process your document with Jupyter.

You can define the processing engine in your YALM header: https://quarto.org/docs/computations/execution-options.html#engine-binding

**Reply by giabaio** [reply]
How weird... I've never had this problem... I guess it must have been some other setup that I haven't managed to recreate yet...
To be clear: your suggestion works. If I add `engine: knitr` to the YAML of the document that didn't work OR if I add a `{r}` chunk to the same document, then the `R` code is processed correctly.

Can the engine binding be done globally, as far as you know?
Thanks!

**Reply by cscheid** [reply]
Adding `engine: knitr` to the YAML of the document _should_ work. If it doesn't for you, please create a minimal reprex and file a new issue. Thanks!

**Reply by giabaio** [reply]
Thank you guys! Yes: adding `engine: knitr` to the YAML of *that* document does work. I'll have to remember that --- I can see in https://github.com/quarto-dev/quarto-cli/issues/3908#issuecomment-1451909190 that the argument was that not many people would use inline `R` code without chunks, which I seem to be doing more than the average user... Oh well... As long as I know what to do!... :-)

**Reply by cderv** [reply]
> I can see in https://github.com/quarto-dev/quarto-cli/issues/3908#issuecomment-1451909190 that the argument was that not many people would use inline R code without chunks, which I seem to be doing more than the average user... Oh well... As long as I know what to do!... :-)

Yes we changed the default behavior because the experience for a non R user, using Quarto only for writing content verbatim without any computation would have a R requirement for writing a simple code example. See original issue https://github.com/quarto-dev/quarto-r/issues/40 or this dummy example

```
Let's define `r` as sum of `x` and `y` as in `r = x + y`
```

This would have trigger inline parsing with knitr without any engine set in YAML. 

So when only inline content like you did and you know you need R, just set the `engine: knitr` - we think it is easier for R user to do that, than non R user to deactivate a default. 

That is why we changed it.

**Reply by giabaio** [reply]
:+1:

**Comment by cderv** [comment, marked as answer]
Use `engine: knitr` in YAML header  (said in https://github.com/quarto-dev/quarto-cli/discussions/4855#discussioncomment-5335059 by @cscheid ) 

For context, when you use inline R code chunk without any R chunk in the document, the [auto binding](https://quarto.org/docs/computations/execution-options.html#engine-binding) won't apply anymore. When you do use also R code chunk, Quarto will auto bind to the knitr computation engine.

**Comment by giabaio** [comment]
Happy with that. Should I close the discussion?
